common/src/main/java/org/apache/apex/common/util/JarHelper.java

```diff
@@ -107,6 +114,73 @@ public String getJar(Class<?> jarClass)
     return jar;
   }
 
+  public String getJar(Class<?> jarClass)
+  {
+    return getJar(jarClass, true);
+  }
+
+  /**
+   * Returns a full path to the jar-file that contains the given class and all full paths to dependent jar-files
+   * that are defined in the property "apex-dependencies" of the manifest of the root jar-file.
+   * If the class is an independent file the method makes jar file from the folder that contains the class
+   * @param jarClass Class
+   * @param makeJarFromFolder True if the method should make jar from folder that contains the independent class
+   * @param addJarDependencies True if the method should include dependent jar files
+   * @return List of names of the jar-files
+   */
+  public List<String> getJars(Class<?> jarClass, boolean makeJarFromFolder, boolean addJarDependencies)
+  {
+    List<String> list = new LinkedList<>();
+    String jar = getJar(jarClass, makeJarFromFolder);
+    if (jar != null) {
+      list.add(jar);
+    }
+
+    if (addJarDependencies) {
+      try {
+        getDependentJarsFromManifest(new JarFile(jar), list);
+      } catch (IOException ex) {
+        logger.warn("Cannot open Jar-file {}", jar);
+      }
+    }
+    return list;
+  }
+
+  /**
+   * Returns a full path to the jar-file that contains the given class and all full paths to dependent jar-files
+   * that are defined in the property "apex-dependencies" of the manifest of the root jar-file.
+   * If the class is an independent file the method makes jar file from the folder that contains the class
+   * @param jarClass Class
+   * @return List of names of the jar-files
+   */
+  public List<String> getJars(Class<?> jarClass)
+  {
+    return getJars(jarClass, true, true);
+  }
+
+  /**
+   * Adds dependent jar-files from manifest to the target list of jar-files
+   * @param jarFile Jar file
+   * @param list List of target jar-files
+   * @throws IOException
+   */
+  public void getDependentJarsFromManifest(JarFile jarFile, List<String> list) throws IOException
+  {
+    String value = jarFile.getManifest().getMainAttributes().getValue(APEX_DEPENDENCIES);
+    if (!StringUtils.isEmpty(value)) {
+      String folderPath = new File(jarFile.getName()).getParent();
```

Comment:
Why should this Path. The value of the parent folder is used as a string for concatenation of strings only.